### PR TITLE
simple-tiles: require python@3.11 to build

### PR DESCRIPTION
**Pull request by Firefishy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

`simple-tiles` builder is not currently compatible with python 3.12+ . This PR pins the build process to using compatible python 3.11. 

`simple-tiles` build failure is currently blocking https://github.com/Homebrew/homebrew-core/pull/154602

### CODE REVIEW

**Comment by Firefishy** [comment]
For completeness I have also PR'ed upstream to `simple-tiles` fix the python version compatibility. https://github.com/propublica/simple-tiles/pull/23

File: Formula/s/simple-tiles.rb [hunk 1]
```diff
@@ -20,14 +20,14 @@ class SimpleTiles < Formula
...
   depends_on "cairo"
   depends_on "gdal"
   depends_on "pango"
 
-  uses_from_macos "python" => :build
-
   def install
-    system "python3", "./waf", "configure", "--prefix=#{prefix}"
+    ENV.prepend_path "PATH", Formula["python@3.11"].libexec/"bin"
+    system "./waf", "configure", "--prefix=#{prefix}"
```
Comment (singingwolfboy) [inline Formula/s/simple-tiles.rb:30]: Better to pass the path to the actual executable you want to use:
```suggestion
    python_exe = Formula["python@3.11"].opt_bin/"python3.11"
    system python_exe, "./waf", "configure", "--prefix=#{prefix}"
```

**Review by singingwolfboy — CHANGES_REQUESTED** [review]

File: Formula/s/simple-tiles.rb (hunk 1 above)
Comment (Firefishy) [reply Formula/s/simple-tiles.rb:30]: That unfortunately does not work, the issue is the project's `Makefile` in the next step also calls `waf` and it uses shebang `/usr/bin/env python`.

Comment (singingwolfboy) [reply Formula/s/simple-tiles.rb:30]: Gotcha, that's reasonable 👍

**Review by singingwolfboy — APPROVED** [review]

**Pull request merged**